ocs_ci/utility/ibm_hci.py

```
@@ -62,6 +78,276 @@ def _load_rack_details(self):
+            )
+            reachable = exit_code == 0
+            if reachable:
+                log.debug(f"BMC {node_ip} is reachable from MGem {rack_ip}")
+            else:
+                log.debug(f"BMC {node_ip} is NOT reachable from MGem {rack_ip}")
+            return reachable
+        except Exception as e:
+            log.debug(f"Ping from MGem to {node_ip} failed with exception: {e}")
+            return False
+
+    def _verify_bmc_connectivity(self):
```

Ruff (0.15.20)

[warning] Too many branches (16 > 12)

(PLR0912)